src/pudl/metadata/fields.py

```diff
@@ -1133,6 +1165,26 @@
         ),
         "unit": "MWh",
     },
+    # TODO[zaneselvans] 2024-04-20: Flesh out description
+    "energy_source": {
```

Comment:
Note also that I shortened some of these to match other categorizations that we have (and also to make this more compact):
- `all_petroleum_products` -> `oil`
- `natural_gas` -> `gas`
- `other_fuel_sources` -> `other`
- `hydro_and_pumped_storage` -> `hydro`
- `unknown_fuel_source` -> `unknown`

Reply:
not blocking but it might be good to make this an encoded table so you can add any description for any of these guys.

Reply:
I think we probably want to avoid encoding many millions of rows if we can, and these values are extracted directly from the renamed column headers so we have complete control over what values they end up containing. But agree we need to document this. Probably this should be a less generic field name, with a description that talks specifically about what's going on in the EIA-930.